app/questionnaire/questionnaire_schema.py

```diff
@@ -603,67 +587,34 @@ def _update_answer_dependencies_for_value_source(
                 )
 
                 for answer_id_for_block in answer_ids_for_block:
-                    self._answer_dependencies_map[answer_id_for_block] |= {
-                        self._get_answer_dependent_for_block_id(
+                    self._answer_dependencies_map[answer_id_for_block].add(
+                        dependent := self._get_dependent_for_block_id(
                             block_id=block_id, answer_id=answer_id
                         )
```

Comment:
Existing but this should be outside the for loop, it is the same for all answers.